RMET-3503 ::: Make `ANALYTICS_COLLECTION_ENABLED` Configurable

OS-ricardomoreirasilva opened the pull request:
## Description
- Aligned with our interpretation of [Firebase's recommendations](https://firebase.google.com/docs/analytics/configure-data-collection?platform=android#disable_data_collection), this property is just added when it's explicitly set to `false`.
- Change the "onload" property to yes, as plugin variables aren't an option with OutSystems. As before, the plugin is initialised when the app is, independently of the preference configuration.

## Context
https://outsystemsrd.atlassian.net/browse/RMET-3503

## Type of changes
- [x] Fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [x] Android
- [x] iOS
- [ ] JavaScript

## Tests
Manual testing was performed, including the different options for the preference:
- No value
- `true` value
- `false` value

## Checklist
- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [x] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

OS-ricardomoreirasilva committed:
chore(ios): add ANALYTICS_COLLECTION_ENABLED to info.plist

Aligned with our interpretation of Firebase's recommendations, this property is just added when it's explicitly set to false.
Change the "onload" property to yes, as plugin variables aren't an option with OutSystems.

References: https://outsystemsrd.atlassian.net/browse/RMET-3503

OS-ricardomoreirasilva committed:
chore(android): add hook to set ANALYTICS_COLLECTION_ENABLED

Aligned with our interpretation of Firebase's recommendations, this property is just added when it's explicitly set to false.
Change the "onload" property to yes, as plugin variables aren't an option with OutSystems.

References: https://outsystemsrd.atlassian.net/browse/RMET-3503

OS-ricardomoreirasilva committed:
chore: add CHANGELOG entry

References: https://outsystemsrd.atlassian.net/browse/RMET-3503

OS-ricardomoreirasilva committed:
chore(android): replace DOMParser with xml2js

References: https://outsystemsrd.atlassian.net/browse/RMET-3503

OS-martacarlos commented on hooks/ios/iOSCopyPreferences.js:
```diff
@@ -7,14 +7,12 @@ module.exports = function (context) {
-    var appNameParser = new ConfigParser(appNamePath);
-    var appName = appNameParser.name();
+    
+    var appName = configParser.name();
```
we can replace all `var` variables with `let`, like it's in `androidCopyPreferences.js`

OS-ricardomoreirasilva replied:
Of course! Can you check again?

OS-ricardomoreirasilva committed:
chore(ios): include pr comments

Replace 'var' with 'let'.

References: https://outsystemsrd.atlassian.net/browse/RMET-3503

OS-ricardomoreirasilva committed:
chore(ios): remove extra file saves

File save is already being done in the end, so there's no need to do it within each change.

References: https://outsystemsrd.atlassian.net/browse/RMET-3503
